src/monet.js

```diff
@@ -177,6 +177,22 @@
         return listFindC(l, fn).run()
     }
 
+    function listContainsC(l, val) {
+        if (l.isNil) {
+            return Return(false)
+        }
+        var h = l.head()
+        return h === val ?
```

Comment:
hmm you surely have thought about this more than I did, but the `if (!a || !b)` frightens me. It makes it that `"" != ""`, because `!""` is `true`. Maybe we should just do `=== undefined` or what it is you have in mind there?